### DISCUSSION TITLE
Does it make sense to develop a web application that does not require SEO with next.js?

### DISCUSSION THREAD

**Discussion by Hashim1995** [vercel/next.js#56356] [Help]
### Summary

In general, I have used react.js for web applications and next.js for websites so far. But in the blogs I have read recently and so on, everyone recommends me to use next.js. Is next.js suitable for developing a huge internal web application? What kind of problems could it cause in the future? What are its pros and cons?

**Comment by icyJoseph** [comment]
Maybe the pre-rendering during build time can get in the way, or that you'd need to opt into "export" mode, which creates a Static HTML output, that obviously doesn't support a lot of Next.js' goodies.

Also, if you go for having a server, the default way, you'd need to maintain your server, and handle potential memory leaks coming from your code.

Other than that, it is pretty much the same thing as doing it with Vite or the old CRA, etc. 

I guess there's more, but those are at top of my head right now. Ask further if that's not enough :)
